core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -443,7 +443,7 @@ object TopicCommand extends Logging {
     print("\tPartition: " + tp.partition)
     print("\tLeader: " + (if(tp.leader.isDefined) tp.leader.get else "none"))
     print("\tReplicas: " + tp.assignedReplicas.mkString(","))
-    print("\tIsr: " + tp.isr.mkString(","))
+    print("\tIsr: " + (if(tp.leader.isDefined) tp.isr.mkString(",") else "none"))
```

Comment:
We still have two problems:

1. This affects both the output using --zookeeper and --bootstrap-server. As I said in the first comment, it is still useful to display the ISR when the leader is offline if we have it available. The problem here is that we do not have an accurate ISR when --bootstrap-server is used.
2. The term "none" may cause confusion. We are trying to convey the fact that we just don't know what the current ISR is.

Maybe we should try to approach this from a different perspective. The root of the problem is that the Metadata API does not return the ISR faithfully when there is no leader. Can we change that? The rest of the information is returned accurately including the replica set and the offline replicas. I cannot think of a good reason for the inconsistency. I'd suggest we change the behavior here: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/MetadataCache.scala#L98. Then we do not need to handle the case specially here.